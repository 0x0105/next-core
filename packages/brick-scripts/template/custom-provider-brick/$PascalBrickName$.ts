import { createProviderClass } from "@easyops/brick-utils";

export interface $PascalBrickName$Params {}

export function $PascalBrickName$(
  params: $PascalBrickName$Params
): Promise<any> {
  return null;
}

customElements.define(
  "$kebab-custom-provider-brick-name$",
  createProviderClass($PascalBrickName$)
);
