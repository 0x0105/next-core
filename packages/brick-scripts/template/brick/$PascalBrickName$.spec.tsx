import React from "react";
import { shallow } from "enzyme";
import { $PascalBrickName$ } from "./$PascalBrickName$";

describe("$PascalBrickName$", () => {
  it("should work", () => {
    const wrapper = shallow(<$PascalBrickName$ />);
    expect(wrapper).toMatchSnapshot();
  });
});
