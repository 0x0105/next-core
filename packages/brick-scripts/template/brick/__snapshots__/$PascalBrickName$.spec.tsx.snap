// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$PascalBrickName$ should work 1`] = `
<div>
  $CONSTANT_PACKAGE_NAME$
   works!
</div>
`;
